homeassistant/components/leviosa_shades/manifest.json

```
@@ -0,0 +1,10 @@
+{
+  "domain": "leviosa_shades",
+  "name": "Leviosa Motor Shades",
+  "codeowners": ["@greg-ha-1990"],
+  "config_flow": true,
+  "dependencies": ["cover"],
+  "documentation": "https://www.home-assistant.io/integrations/leviosa_shades",
+  "iot_class": "local_push",
```

```suggestion
  "iot_class": "assumed_state",
```

This is why I was confused by no state in your integration. If you have a stateless integration (as you described in the reply, added below), your iot class is assumed state. See https://developers.home-assistant.io/docs/creating_integration_manifest/#iot-class.

> This is a wifi to remote device, it can't actively know the state of the cover because there is no network connection between them, just as there is no way to ask the TV remote which channel is playing on the TV right now. We can only save the cached state values when actively controlling a wifi to IR device, and besides, cover can only be controlled by this IR device, so there's no ambiguity about it.

---

感谢您的建议，我之前忽略了assumed_state的选项，已更正，同时重新添加了以下属性：
```python
@property
def assumed_state(self):
    """Indicate that we do not go to the device to know its state."""
    return True
```

---

Suggestion accepted, thanks.